### PR TITLE
✨feat added environment variables and sync.yml

### PR DESCRIPTION
**Pull request by Yanyutin753**
## Description

✨feat added environment variables and sync.yml

---
### Changelog Entry

#1793

### Fixed
✨feat added environment variables 
✨feat added sync.yml

### Changed

| Section | Content | Description |
|:--------|:---------------------|:------------|
| Images  | IMAGE_GENERATION_ENGINE | Select the image generation engine. Options: `openai`, `ollama`, `comfyul`. |
|         | IMAGE_GENERATION_MODEL | Selects the image model. |
|         | IMAGE_SIZE | Sets the image size. Default is `512x512`. |
|         | IMAGE_STEPS | Sets the image steps. Default is `50`. |
| RAG | PDF_EXTRACT_IMAGES | Toggles whether to enable OCR for recognizing PDF. |
| Additions | sync.yml | Added `sync.yml` for automatic synchronization and updates of the latest code for forked repositories using github action. 

### Removed
NO Removed, Only Add

### Verify
I have self-tested and it passed.

image

image

image

### CODE REVIEW

File: backend/apps/images/main.py [hunk 1]
```diff
@@ -32,11 +32,15 @@
 from config import (
     SRC_LOG_LEVELS,
     CACHE_DIR,
+    IMAGES_GENERATION_ENGINE,
```
Comment (tjbck) [inline backend/apps/images/main.py:35]: Should be `IMAGE_GENERATION_ENGINE`

File: backend/apps/images/main.py [hunk 2]
```diff
@@ -32,11 +32,15 @@
...
     ENABLE_IMAGE_GENERATION,
     AUTOMATIC1111_BASE_URL,
     COMFYUI_BASE_URL,
     IMAGES_OPENAI_API_BASE_URL,
     IMAGES_OPENAI_API_KEY,
+    IMAGES_MODEL,
```
Comment (tjbck) [inline backend/apps/images/main.py:41]: Should be more explicit `IMAGE_GENERATION_MODEL`

Comment (Yanyutin753) [reply backend/apps/images/main.py:41]: ok,i will recompose it

File: backend/apps/images/main.py (hunk 1 above)
Comment (Yanyutin753) [reply backend/apps/images/main.py]: ok,Thank you

**Comment by tjbck** [comment]
LGTM, Thanks!

**Comment by Yanyutin753** [comment]
> LGTM, Thanks!

hhh, Thank the author!

**Comment by BrandXX** [comment]
Thank you

**Pull request merged**